### ISSUE TITLE
Getting test predict() to line up with dataframe

### ISSUE THREAD

**Issue by pcgm-team** [jdb78/pytorch-forecasting#1459]
After we have a trained model and get the predictions, how can we retrieve the predictions on a test set and get it to line up with index in a dataframe?

I've attempted the following code:
```
new_raw_predictions = tft.predict(test_data, mode="prediction", return_index = True)
results_df = pd.DataFrame(new_raw_predictions.output.detach().cpu().numpy(), columns = ['step1', 'step2', 'step3'])
res = pd.merge(new_raw_predictions.index,results_df, right_index=True, left_index=True)
```

This is great and lines up the predictions with the timestamps. However, rather than having predictions for each row of `test_data`, I get a much smaller set of randomly dispersed rows.

Related issues: #734 #961 #1008 #1316

**Comment by pcgm-team** [comment]
Ok actually reading the docs, the issue may have been that in the test_data I had set `predict=True`; which limits the output to 1 per group. I'll see if this fixes it.
